pubsub/subscriptions/sync_pull.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriptions
+
+// [START pubsub_subscriber_sync_pull]
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+
+	"cloud.google.com/go/pubsub"
+)
+
+func pullMsgsSync(w io.Writer, projectID, subName string, topic *pubsub.Topic) error {
+	// projectID := "my-project-id"
+	// subName := projectID + "-example-sub"
+	// topic of type https://godoc.org/cloud.google.com/go/pubsub#Topic
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("pubsub.NewClient: %v", err)
+	}
+
+	var mu sync.Mutex
+	received := 0
+	sub := client.Subscription(subName)
+
+	// Turn on synchronous mode. This makes the subscriber use the Pull RPC rather
+	// than the StreamingPull RPC, which is useful for guaranteeing MaxOutstandingMessages,
+	// the max number of messages the client will hold in memory.
+	sub.ReceiveSettings.Synchronous = true
+	sub.ReceiveSettings.MaxOutstandingMessages = 10
+
+	cctx, cancel := context.WithCancel(ctx)
+	err = sub.Receive(cctx, func(ctx context.Context, msg *pubsub.Message) {
+		mu.Lock()
+		defer mu.Unlock()
+		fmt.Fprintf(w, "Got message :%q\n", string(msg.Data))
+		_ = msg // TODO: handle message.
+		msg.Ack()
+		if received++; received == 10 {
```

Comment:
Optional: I might put the `received++` on its own line. That's the important part and `++` doesn't return a value. Inline `if` declarations are normally used for declarations of stuff used in the `if` statement (like errors and `got`).